doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,629 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2021-02-23
+last updated: 2021-02-24
+issue id: #653
+---
+
+# Quake Mode
+
+## Abstract
+
+Many existing terminals support a feature whereby a user can press a keybinding
+anywhere in the OS, and summon their terminal application. Oftentimes the act of
+summoning this window is accompanied by a "dropdown" animation, where the window
+slides in to view from the top of the screen. This global summon action is often
+referred to as "quake mode", a reference to the videogame Quake who's console
+slid in from the top.
+
+This spec addresses both of the following two issues:
+* "Quake Mode" ([#653])
+* "Minimize to tray" ([#5727])
+
+Readers should make sure to have read the [Process Model 2.0 Spec], for
+background on Monarch and Peasant processes.
+
+## Background
+
+### Inspiration
+
+For an example of the original Quake console in action, take a look at the
+following video (noisy video warning): [Quake 3 sample]. Additionally, plenty of
+existing terminal emulators support similar functionality:
+
+* **Tilda** allows the user to specify different keys to summon the window on
+  different monitors.
+* **Guake** alternatively allows the user to either sumon the terminal window to
+  a specific monitor, or whichever monitor the mouse is on. Guake only allows
+  one single instance, so pressing the global hotkey always summons the same
+  instance.
+
+### User Stories
+
+The original quake mode thread (#653) is absolutely _filled_ with variations on
+how users want to be able to summon their terminal windows. These include, but
+are not limited to:
+
+* **Story A** Press a hotkey anywhere to activate the single Terminal window
+  wherever it was
+* **Story B** Press a hotkey anywhere to activate the single Terminal window _on
+  the current monitor_. If it wasn't previously on that monitor, move it there.
+* **Story C** When the Terminal is summoned using the hotkey, have it "slide in"
+  from the top. Similarly, slide out on deactivate.
+* **Story D** <kbd>Ctrl+1</kbd> to activate the terminal on monitor 1,
+  <kbd>Ctrl+2</kbd> to activate the terminal on monitor 2.
+* **Story E** Variable dropdown speed
+* **Story F** Minimize to tray, press a hotkey to activate the terminal window
+  (#5727)
+* **Story G** Terminal doesn't appear in alt+tab view, press a hotkey to

```

Comment:
"Story G" doesn't get addressed by this spec, I think. This is also specifically how iTerm2 works and I feel like that's the mode that speaks most to me.

Reply:
Isn't that "minimize to tray"?

Reply:
G could be separate from F. It just feels like a next step to it where you could add a setting to remove it from alt+tab, possibly regardless of whether it's single mode or not, or only if minimize to tray is enabled.